lib/helpers/backend/valid-isbn.ts

```diff
@@ -0,0 +1,3 @@
+export function isValidISBN(isbn: string) {
+  return (isbn.length == 13 && isbn.match(/^[0-9]+$/));
```

Comment:
This doesn't meet the requirements for a valid ISBN since the last digit is a checksum. I'd recommend using a package (such as https://www.npmjs.com/package/isbn) for this so we don't have to own the logic.

Reply:
Also, not all of the campus store ISBNs are even valid ISBNs from what I can tell (see `281000000862B`), so maybe we don't do checking at all since this API is just a GET operation?

Reply:
An ISBN 281000000862B indicates that there is not textbook available on the bookstore (this is what the bookstore returns when no textbooks are found). That ISBN will not be part of the database.

I tried the isbn package, and it works well. I think it is important to have a front check, so we reject any invalid requests prior to making the call.

Reply:
@alkoasma What about these cases?

```
{
  "isbn": "281000000862B",
  "name": "SIMULATION STRATEGIC MANAGEMENT - AUTO INDUSTRY MERGER AND ACQUISITION",
  "program": "McMaster Undergrad and Graduate",
  "term": "Fall 2021",
  "deptName": "BUSADMIN",
  "deptAbv": "BUSADMIN",
  "course": "I605",
  "section": "DAY C04",
  "price": "Not Available",
  "image": "https://campusstore.mcmaster.ca/cgi-mcm/ws/getTradeImage.pl?isbn=281000000862B"
},
{
  "isbn": "281000000881B",
  "name": "AC SALES PROFILE ASSESSMENT",
  "program": "McMaster Undergrad and Graduate",
  "term": "Fall 2021",
  "deptName": "BUSADMIN",
  "deptAbv": "BUSADMIN",
  "course": "I605",
  "section": "DAY C04",
  "price": "Not Available",
  "image": "https://campusstore.mcmaster.ca/cgi-mcm/ws/getTradeImage.pl?isbn=281000000881B"
},
{
  "isbn": "281000000882B",
  "name": "AC QUALITY SERVICE ASSESSMENT",
  "program": "McMaster Undergrad and Graduate",
  "term": "Fall 2021",
  "deptName": "BUSADMIN",
  "deptAbv": "BUSADMIN",
  "course": "I605",
  "section": "DAY C04",
  "price": "Not Available",
  "image": "https://campusstore.mcmaster.ca/cgi-mcm/ws/getTradeImage.pl?isbn=281000000882B"
},
```

Reply:
Great catch. I went through a couple of codes that start with "2810000008" on the campus store site. They are each related to something specific being conveyed or are related to in-class content.

Will confirm this and create one entry for each of those codes in the database.

Will have to remove the validation entirely too.

Thanks Caleb.